### PR TITLE
Add Block#isValidTool

### PR DESCRIPTION
**Pull request by Machine-Maker**
As talked about in the discord

### CODE REVIEW

**Comment by FranKusmiruk** [comment]
Just an idea, but I would rename  the method to be `isBreakable(ItemStack)` and also an additional method `isBreakable` without arguments to match the style of `breakNaturally`

**Comment by kickash32** [comment]
isBreakable returning false implies you cannot break it at all like bedrock. I like the idea with following the style of breakNaturally though.

File: Spigot-API-Patches/0218-Add-Block-isValidTool.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+     }
++
++    /**
++     * Checks if the itemstack is a valid tool to
++     * break the block with
++     *
++     * @param itemStack The (tool) itemstack
++     * @return whether the block will drop items
++     */
++    boolean isValidTool(ItemStack itemStack);
```
Comment (Proximyst) [inline Spigot-API-Patches/0218-Add-Block-isValidTool.patch]: Perhaps we should use a tristate here instead? I.e. either you can break it as it's supposed to, can break but not ideal tool, or cannot break at all. (cc @kickash32)

**Review by Proximyst — CHANGES_REQUESTED** [review]

**Comment by kickash32** [comment]
I believe there is also issue with the implementation, this only checks if the drops are non zero, however for certain actions like breaking ice with a non silk touch pickaxe, you could argue that it is a valid tool. Maybe add this information to the docs that this does not behave like some people may expect it to. Maybe theres something in adventure mode we could use?

File: Spigot-API-Patches/0218-Add-Block-isValidTool.patch (hunk 1 above)
Comment (kickash32) [reply Spigot-API-Patches/0218-Add-Block-isValidTool.patch]: Not exactly opposed to it but maybe that should be a *different* method that returns a 3 valued enum. The original purpose of this API addition was to have a create a simple API call which would help reduce code complexity for API consumers, so lets not lose sight of that.

**Comment by Machine-Maker** [comment]
k, rebased.

**Review by electronicboy — DISMISSED** [review]

**Pull request merged**